### PR TITLE
Added bubblesort in bash

### PR DESCRIPTION
**Pull request by Ariana1729**
More bash algos!

```
(( tmp = arr[(( j + 1 ))] ))
(( arr[(( j + 1 ))] = arr[j] ))
```

these look pretty crippled tho not sure how to make it clearer tbh

### CODE REVIEW

File: contents/bubble_sort/code/bash/bubble_sort.bash [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+bubble_sort() {
+    local i
+    local j
+    local tmp
+    local len
+    local arr
+    arr=("$@")
+    (( len = ${#arr[@]} ))
+
+    for i in $(seq 0 $(( len - 1 ))); do
```
Comment (berquist) [inline contents/bubble_sort/code/bash/bubble_sort.bash]: Don't shell out when you can use a built-in:
```bash
for ((i = 0; i <= len - 1; i++)); do
```

Comment (Ariana1729) [reply contents/bubble_sort/code/bash/bubble_sort.bash]: ah ok

**Comment by jiegillet** [comment]
I just want to say that I really appreciate the bash submissions and reviews, I'm learning a lot and it makes me want to learn proper bash instead the usual SO copy/paste that I hate ^^

**Comment by Liikt** [comment]
I don't think this is how one would typically write bash.

```bash
#!/usr/bin/env bash
bubble_sort() {
    local i
    local j
    local tmp
    local len
    local arr
    arr=("$@")
    len=${#arr[@]}
    for i in $(seq 0 $(calc -e $len-1)); do
            for j in $(seq 0 $(calc -e $len-2)); do
                if [ ${arr[j]} -gt ${arr[j+1]} ]; then
                        tmp=${arr[j+1]}
                        arr[j+1]=${arr[j]}
                        arr[j]=$tmp
                fi
            done
    done
    echo ${arr[*]}
}
arr=(1 45 756 4569 56 3 8 5 -10 -4)
echo "Unsorted array: ${arr[*]}"
tmp=$(bubble_sort "${arr[@]}")
echo "Sorted array: ${tmp[*]}"```

This is how i would write bubblesort in bash

**Comment by Ariana1729** [comment]
Hmm I was refering to [this pull request](https://github.com/algorithm-archivists/algorithm-archive/pull/497) for which one to choose

**Comment by berquist** [comment]
@Liikt It's not how one would normally write bash, because we have the bad habit of shelling out (`seq`, `calc`) when bash internals should be used instead. Using `[...]` is also bad; that's the original Bourne shell, not the Bourne-again shell.

If this were pure `sh`, then yes, you would need to call out to other programs.

**Comment by berquist** [comment]
@Ariana1729 I'll look over this again tonight my time.

**Comment by Liikt** [comment]
I mean sure. I just have never seen even bash written like that. But then again I can't say i have seen that much bash at all. I personally still prefer my style but I can agree that it is more sh than bash. What I found is that the current code looks a bit like lisp :D

**Comment by berquist** [comment]
:+1:

**Pull request merged**